libmate-desktop/mate-colorsel.c

```diff
@@ -1743,15 +1687,16 @@ key_press (GtkWidget   *invisible,
            GdkEventKey *event,
            gpointer     data)
 {
-  GdkDevice *device;
+  GdkDevice *device, *pointer_device;
   GdkDisplay *display = gtk_widget_get_display (invisible);
   GdkScreen *screen = gdk_event_get_screen ((GdkEvent *)event);
   guint state = event->state & gtk_accelerator_get_default_mod_mask ();
   gint x, y;
   gint dx, dy;
 
   device = get_device (display);
-  gdk_device_get_position (device, NULL, &x, &y);
+  pointer_device = gdk_device_get_associated_device (device);
```

Comment:
That is *not* working, as `get_device()` already returns the display's seat's default pointer, [`gdk_device_get_associated_device()`](https://docs.gtk.org/gdk3/method.Device.get_associated_device.html) will give you the display's seat's default keyboard -- which doesn't have a position.

So either remove this and make `pointer_device = device`, or replace the `get_device()` call with `gdk_event_get_device((GdkEvent*) event)` so you effectively get the pointer associated with the keyboard triggering the event.

```diff
diff --git a/libmate-desktop/mate-colorsel.c b/libmate-desktop/mate-colorsel.c
index 44b6074..6bcd442 100644
--- a/libmate-desktop/mate-colorsel.c
+++ b/libmate-desktop/mate-colorsel.c
@@ -1567,17 +1567,6 @@ make_picker_cursor (GdkScreen *screen)
   return cursor;
 }
 
-static GdkDevice *get_device (GdkDisplay *display)
-{
-  GdkSeat   *seat;
-  GdkDevice *device;
-
-  seat = gdk_display_get_default_seat (display);
-  device = gdk_seat_get_pointer (seat);
-
-  return device;
-}
-
 static void
 grab_color_at_mouse (GdkScreen *screen,
 		     GdkDevice *device,
@@ -1688,13 +1677,12 @@ key_press (GtkWidget   *invisible,
            gpointer     data)
 {
   GdkDevice *device, *pointer_device;
-  GdkDisplay *display = gtk_widget_get_display (invisible);
   GdkScreen *screen = gdk_event_get_screen ((GdkEvent *)event);
   guint state = event->state & gtk_accelerator_get_default_mod_mask ();
   gint x, y;
   gint dx, dy;
 
-  device = get_device (display);
+  device = gdk_event_get_device ((GdkEvent *) event);
   pointer_device = gdk_device_get_associated_device (device);
   gdk_device_get_position (pointer_device, NULL, &x, &y);
 
```

Reply:
done